internal/common/thrift_util.go

```diff
@@ -92,15 +71,13 @@ func TListDeserialize(ts []thrift.TStruct, b []byte) (err error) {
 
 // IsUseThriftEncoding checks if the objects passed in are all encoded using thrift.
 func IsUseThriftEncoding(objs []interface{}) bool {
-	// NOTE: our criteria to use which encoder is simple if all the types are serializable using thrift then we use
-	// thrift encoder. For everything else we default to gob.
-
 	if len(objs) == 0 {
 		return false
 	}
-
-	for i := 0; i < len(objs); i++ {
-		if !IsThriftType(objs[i]) {
+	// NOTE: our criteria to use which encoder is simple if all the types are serializable using thrift then we use
+	// thrift encoder. For everything else we default to gob.
```

Comment:
~~gob~~ json (we don't use gob anywhere), but this is just a move so it's probably fine.